services/QuillLMS/client/app/bundles/PremiumHub/components/usage_snapshots/shared.ts

```diff
@@ -0,0 +1,360 @@
+            headers: ['Activity', 'Activities completed'],
+            comingSoon: true
+          },
+          {
+            label: 'Most completed activities',
+            type: RANKING,
+            queryKey: 'most-completed-activities',
+            headers: ['Activity', 'Activities completed'],
+            comingSoon: true
+          }
+        ]
+      },
+    ]
+  },
+  {
+    name: CLASSROOMS,
+    className: 'classrooms',
+    itemGroupings: [
+      {
+        className: "counts",
+        items: [
+          {
+            label: 'Active classrooms',
+            singularLabel: 'Active classroom',
+            size: SMALL,
+            type: COUNT,
+            queryKey: 'active-classrooms'
+          },
+          {
+            label: 'Average active classrooms per teacher',
+            size: SMALL,
+            type: COUNT,
+            queryKey: 'average-active-classrooms-per-teacher',
+            comingSoon: true
+          },
+          {
+            label: 'Classrooms created',
+            singularLabel: 'Classroom created',
+            size: SMALL,
+            type: COUNT,
+            queryKey: 'classrooms-created',
+            comingSoon: true
+          },
+          {
+            label: 'Average active students per classroom',
+            size: SMALL,
+            type: COUNT,
+            queryKey: 'average-active-students-per-classroom',
+            comingSoon: true
+          },
+        ]
+      },
+      {
+        className: 'rankings',
+        items: [
+          {
+            label: 'Most active grades',
+            type: RANKING,
+            queryKey: 'most-active-grades',
+            headers: ['Grade', 'Activities completed'],
+            comingSoon: true
+          },
+        ]
+      }
+    ]
+  },
+  {
+    name: SCHOOLS,
+    className: 'schools',
+    itemGroupings: [
+      {
+        className: 'ranking',
+        items: [
+          {
+            label: 'Most active schools',
+            type: RANKING,
+            queryKey: 'most-active-schools',
+            headers: ['School', 'Activities completed']
+          },
+        ]
+      }
+    ]
+  }
+]
+
+export function unorderedArraysAreEqual(arr1, arr2) {
+  if (arr1?.length !== arr2?.length) {
+    return false;
+  }
+
+  // Create a copy of the second array
+  const arr2Copy = [...arr2];
+
+  for (let i = 0; i < arr1.length; i++) {
+    const obj1 = arr1[i];
+    const obj2Index = arr2Copy.findIndex(obj => objectsAreEqual(obj, obj1));
+
+    if (obj2Index === -1) {
+      return false;
+    }
+
+    // Remove the matched object from the copy
+    arr2Copy.splice(obj2Index, 1);
+  }
+
+  return true;
+}
+
+function objectsAreEqual(obj1, obj2) {
+  const obj1Keys = Object.keys(obj1);
+  const obj2Keys = Object.keys(obj2);
+
+  if (obj1Keys.length !== obj2Keys.length) {
+    return false;
+  }
+
+  for (let key of obj1Keys) {
+    if (obj1[key] !== obj2[key]) {
+      return false;
+    }
+  }
+
+  return true;
+}
```

Comment:
It kind of feels that these are generalizable enough that we might want them shared at an even higher level of the directory hierarchy.  Not a problem leaving them here, but just a thought.